### PR TITLE
Sign quote

### PR DESCRIPTION
**Pull request by aguschirico**

### CODE REVIEW

File: providers/liquidity.go [hunk 1]
```diff
@@ -116,7 +120,24 @@ func (lp *LocalProvider) SetLiquidity(value *big.Int) {
 	lp.liquidity = value
 }
 
-func (lp *LocalProvider) SignHash(hash []byte) ([]byte, error) {
+func (lp *LocalProvider) RefundLiquidity(hash string, value *big.Int) error {
```
Comment (javicid) [inline providers/liquidity.go:123]: I would make hash a []byte for consistency.

Also, what's the point of passing the value here if the provider already has it?

File: providers/liquidity.go [hunk 2]
```diff
@@ -116,7 +120,24 @@ func (lp *LocalProvider) SetLiquidity(value *big.Int) {
...
+	val, ok := lp.retainedQuotes[hash]
+	if !ok || val.Int64() != value.Int64() {
+		return fmt.Errorf("invalid quote or value. hash: %v. value: %v", hash, value)
+	}
+	lp.liquidity.Add(lp.liquidity, &val)
+	delete(lp.retainedQuotes, hash)
+	return nil
+}
+
+func (lp *LocalProvider) SignQuote(hash []byte, reqLiq *big.Int) ([]byte, error) {
```
Comment (javicid) [inline providers/liquidity.go:133]: Or make hash here a string

**Review by javicid — CHANGES_REQUESTED** [review]

File: providers/liquidity.go (hunk 1 above)
Comment (aguschirico) [reply providers/liquidity.go:123]: ok changed. The provider stores a map with key value pairs of quotehash|value so when refunding the value of a certain quote, the provider verifies that the quote exists in the map and the value matches the value retained for it.

File: providers/liquidity.go (hunk 2 above)
Comment (aguschirico) [reply providers/liquidity.go:134]: fixed.

**Review by javicid — APPROVED** [review]

**Pull request merged**